symengine/pow.cpp

```diff
@@ -89,8 +94,19 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
     }
     if (eq(*b, *one))
         return a;
-    if (eq(*a, *zero))
-        return zero;
+
+    if (eq(*a, *zero)) {
+        if (is_a_Number(*b)
+            and rcp_static_cast<const Number>(b)->is_positive()) {
+            return zero;
+        } else if (is_a_Number(*b)
+                   and rcp_static_cast<const Number>(b)->is_negative()) {
+            return ComplexInf;
+        } else {
+            return make_rcp<const Pow>(a, b);
```

Comment:
I don't know whether this is desired. In SymPy, assumptions can check whether `b > 0` or not. In symengine we cannot and therefore keeps `0**sqrt(2)` as it is.